crates/bevy_ecs/src/entity/mod.rs

```diff
@@ -886,8 +886,10 @@ impl Entities {
 
     /// Destroy an entity, allowing it to be reused.
     ///
+    /// Returns the `Option<EntityLocation>` of the entity or `None` if the `entity` was not present.
+    ///
     /// Must not be called while reserved entities are awaiting `flush()`.
-    pub fn free(&mut self, entity: Entity) -> Option<EntityLocation> {
+    pub fn free(&mut self, entity: Entity) -> Option<Option<EntityLocation>> {
```

Comment:
`Option<Option<T>>` can be hard to reason about.  Does anyone care about `Some(None)` vs `None`, or could we flatten this and return a plain `Option<EntityLocation>`?  It looks like the only place this is called changed from `.expect()` to `.expect().expect()`, so it doesn't seem to care.  

Otherwise, it might be worth returning a `Result` with some new `enum` that gives names to the two kinds of `None`.